File: rust/src/models/rate_pricing.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct RatePricing {
    pub rate: rust_decimal::Decimal,
}

impl RatePricing {
    pub fn new(rate: rust_decimal::Decimal) -> Self {
        Self { rate }
    }
}

File: typescript/src/models/ratePricing.ts
// this file is @generated

export interface RatePricing {
  rate: string;
}

export const RatePricingSerializer = {
  _fromJsonObject(object: any): RatePricing {
    return {
      rate: object["rate"],
    };
  },

  _toJsonObject(self: RatePricing): any {
    return {
      rate: self.rate,
    };
  },
};
